src/plugins/telemetry/server/fetcher.ts

```diff
@@ -186,13 +186,16 @@ export class FetcherTask {
     failureCount,
     failureVersion,
     currentVersion,
+    lastReported,
   }: TelemetryConfig) {
     if (failureCount > 2 && failureVersion === currentVersion) {
       return false;
     }
 
     if (telemetryOptIn && telemetrySendUsageFrom === 'server') {
-      if (!this.lastReported || Date.now() - this.lastReported > REPORT_INTERVAL_MS) {
+      // Check both: in-memory and SO-driven value.
+      // This will avoid the server retrying over and over when it has issues with storing the state in the SO.
+      if (isReportIntervalExpired(this.lastReported) && isReportIntervalExpired(lastReported)) {
```

Comment:
This implies the 2nd risk I listed in the description of the PR. I hope we're OK with it.